src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/ImageFileService.java

```diff
@@ -58,46 +58,79 @@ public void setDao(D dao) {
 	}
 
 	/**
+	 * The default value used for the creation of thumbnails in pixels
+	 */
+	private final Integer DEFAULT_THUMBNAIL_SIZE = 100;
+
+	/**
+	 * @param uploadedImage
+	 * @return
+	 * @throws Exception
+	 */
+	public ImageFile uploadImageFile(MultipartFile uploadedImage) throws Exception {
+
+		if (uploadedImage == null || uploadedImage.isEmpty()) {
+			final String errMsg = "Upload failed. Image " + uploadedImage + " is empty.";
```

Comment:
NullPointer will never happen here as the evaluation of `if (uploadedImage == null || uploadedImage.isEmpty())` would stop after the first check if uploadedImage would be null :wink: